R/teal_data_utils.R

```diff
@@ -42,35 +42,37 @@ NULL
 .subset_teal_data <- function(data, datanames) {
   checkmate::assert_class(data, "teal_data")
   checkmate::assert_class(datanames, "character")
-  datanames_corrected <- intersect(datanames, ls(data@env))
-  dataname_corrected_with_raw <- intersect(c(datanames, sprintf("%s._raw_", datanames)), ls(data@env))
-
+  datanames_corrected <- intersect(datanames, .teal_data_ls(data))
+  datanames_corrected_with_raw <- c(datanames_corrected, ".raw_data")
   if (!length(datanames)) {
     return(teal_data())
   }
 
   new_data <- do.call(
     teal.data::teal_data,
     args = c(
-      mget(x = dataname_corrected_with_raw, envir = data@env),
+      mget(x = datanames_corrected_with_raw, envir = teal.code::get_env(data)),
       list(
         code = gsub(
           "warning('Code was not verified for reproducibility.')\n",
           "",
-          teal.data::get_code(data, datanames = dataname_corrected_with_raw),
+          teal.data::get_code(data, datanames = datanames_corrected_with_raw),
           fixed = TRUE
         ),
```

Comment:
a nasty coupling here
to be addressed outside of this PR

Reply:
could be avoided with:
- moving this to teal.data - see: https://github.com/insightsengineering/teal/pull/1342/files#r1772857752
- add argument to `teal.data::get_code` (in particular: not `teal.code::get_code()`

Reply:
> moving this to teal.data - see: https://github.com/insightsengineering/teal/pull/1342/files#r1772857752
 
 I don't see the possibility to move this to `teal.data`. `datanames` argument is fed by `module$datanames` + `teal_transform_module@datanames`.

>  add argument to teal.data::get_code (in particular: not teal.code::get_code()

What argument in `get_code`? I don't think that `get_code` needs any argument. Theoretically, `get_code(<teal_data>, datanames = <names of the objects>)` accepts any name even those prefixed by `.`.

Reply:
I was thinking about the following: `get_code(<teal_data>, datanames = ..., include_warnings = FALSE)`.

The warning code is added inside `teal.data` so `teal.data` should be also responsible for removing it. We should ask ourselves a question: what if the warning message got changed for some reason? Right now we would have to process this change in two repos. We should avoid such coupling. This is the type of the things that I would always trying to point out.
If you don't agree on extending `get_code()` - maybe the whole `subset_teal_data()` should live in teal.data?

Reply:
Maybe warning shouldn't be produced at all in `qenv`? `teal` adds `stopifnot` anyway so it will throw an error anyway id datasets are not reproducible.

Reply:
Even better! qenv should return an analysis / datasets source code. This warning belongs to neither of them.

Reply:
here I'm removing warning message:
https://github.com/insightsengineering/teal.data/pull/334